### ISSUE TITLE
Use docker compose file for CI stuff

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#1001]
In #999, I, with the help of @malteos, created a docker image for CourtListener. It's pretty great! Now we just need to use it in CircleCI.

That'll prove a little bit difficult, but probably not too bad. The only manual steps that currently remain are pulling CourtListener and CL-Solr code from git, and then installing it into the various volumes. If I can automate that, this should be done. 

Oh, and we need to automate migrations too.

cc: @davidxia

**Comment by mlissner** [comment]
There seems to be a fair bit of bad documentation on this topic, but this looks simple and good: https://medium.com/@einenlum/testing-your-app-with-docker-compose-on-circleci-ffd4cc8e2cfa

**Comment by mlissner** [comment]
So I started looking into the CircleCI stuff, but got annoyed that it was too complicated. Simultaneously, I'd been hearing good things about Github Actions, and they came out of beta, so I gave them a whirl. It also seemed nicer to start with a blank slate, and the performance of the CircleCI website had been annoying me. 

It took a pretty big effort, but I got Github Actions working with our docker compose file. Along the way, I:

 - [x]: Deleted our old testing docker image from docker hub and from our codebase (goodbye `.circleci` directory). This is a really nice, big step because this was established before we had our docker compose file, and needed to stay in sync with it, which was annoying. It also had a *lot* of complexity that we had practically copy/pasted to our docker compose file, so all of that is just...*poof*. Gone.

 - [x]: Added a code formatting tool called black that also uses Github Actions. This will ensure that the code is well formatted (I hope), and with any luck will prevent us from having discussions about formatting in the future.

 - [x]: Cleaned up some inconsistencies in our `05-private.example` file so that it's good enough to use without change on a testing or development machine.

Remaining todos that come to mind:

 - [ ] Update our DEVELOPING.md file
 - [ ] Get updates going from Github to Slack, if we care (I'm not sure we do).
